src/path/jit.rs

```diff
@@ -266,6 +266,8 @@ mod test {
                 ],
             ),
             (".foo", vec![BorrowedSegment::Field("foo".into())]),
+            (".a-b", vec![BorrowedSegment::Field("a-b".into())]),
+            (".-a-b", vec![BorrowedSegment::Invalid]),
```

Comment:
What happens if there is a trailing `-`?

Reply:
It will parse correctly (I just added a test case for it). The leading hyphen case fails because the JitState is `EventRoot` for the first parsed char and this PR only adds hyphen support when JitState is `Field`.

I've come around to thinking it's better to add hyphen support for all `JitState` however, because that's more consistent. If a user truly wants to start a key with a hyphen, who am I to prevent it?